fml/task_runner.h

```diff
@@ -16,6 +16,10 @@ namespace fml {
 
 class MessageLoopImpl;
 
+/// To avoid data races between the platform thread posting sync tasks to the
+/// raster thread and the raster thread merging.
+extern std::mutex g_thread_merging_lock;
```

Comment:
When adding a mutex, I would ask myself what the critical section is. From the documentation, it is not clear what that is.